src/app/core/store/collections/collections.effects.ts

```diff
@@ -77,6 +77,8 @@ export class CollectionsEffects {
   @Effect()
   public getNames$: Observable<Action> = this.actions$.pipe(
     ofType<CollectionsAction.GetNames>(CollectionsActionType.GET_NAMES),
+    withLatestFrom(this.store$.select(selectCollectionNames)),
+    filter(([action, collectionNames]) => isNullOrUndefined(collectionNames)),
```

Comment:
I am not sure if we want to work with the old data when we check for duplicates on new collection creation. Such performance fixes can be implemented after we have push notifications. Or is it called too many times?